pkg/apis/cluster/v1alpha1/migrations/migrations.go

```diff
@@ -141,26 +142,34 @@ func MigrateMachinesv1Alpha1MachineToClusterv1Alpha1MachineIfNecessary(
 	kubeClient kubernetes.Interface,
 	providerData *cloudprovidertypes.ProviderData) error {
 
-	err := client.Get(ctx, types.NamespacedName{Name: machines.CRDName}, &apiextensionsv1beta1.CustomResourceDefinition{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			klog.Infof("CRD %s not present, no migration needed", machines.CRDName)
-			return nil
+	for try := 0; try < 15; try++ {
```

Comment:
There *has* to be a better way to wait. I'm somewhat sure that there are helpers for waiting and exponential backoffs and all that stuff.